llvmlite/ir/module.py

```diff
@@ -42,22 +42,32 @@ def _fix_di_operands(self, operands):
             fixed_ops.append((name, op))
         return fixed_ops
 
-    def add_metadata(self, operands):
+    def add_metadata(self, operands, *, self_ref=False):
         """
         Add an unnamed metadata to the module with the given *operands*
         (a sequence of values) or return a previous equivalent metadata.
         A MDValue instance is returned, it can then be associated to
         e.g. an instruction.
+
+        A self referential metadata entry has itself as the first
+        operand to ensure uniquencess. These references are never
+        cached.
```

Comment:
Can you add few words to relate this to the `self_ref` argument? e.g. the `self_ref` creates a self referential metadata by prepending itself to the operands?.

Reply:
I extended the description.